### PR TITLE
Add helper methods to check whether feed or url is a query feed or url

### PR DESCRIPTION
**Pull request by kmws**
The code duplication drove me nuts.

### CODE REVIEW

**Comment by tsipinakis** [comment]
We have other types like `filter` and `exec` feeds, perhaps an enum would be more appropriate here?

```c
f.feed_type() == FEED_RSS
f.feed_type() == FEED_QUERY
f.feed_type() == FEED_EXEC
...
```

File: test/rss.cpp [hunk 1]
```diff
@@ -711,4 +711,21 @@ TEST_CASE("If item's <title> is empty, try to deduce it from the URL",
 	REQUIRE(feed->items()[4]->title() == "Alternate link isn't first");
 }
 
+TEST_CASE("rss_feed::is_query_feed()", "[rss]")
```
Comment (Minoru) [inline test/rss.cpp]: Please provide a description here. If you wanted me to write that method and you could only use one sentence to explain it to me, what would it be? :)

File: test/utils.cpp [hunk 2]
```diff
@@ -707,3 +707,11 @@ TEST_CASE(
 		REQUIRE_FALSE(utils::is_valid_color(input));
 	}
 }
+
+TEST_CASE("is_query_url()", "[utils]")
```
Comment (Minoru) [inline test/utils.cpp]: Ditto: please provide a description for the test.

**Comment by Minoru** [comment]
@tsipinakis Maybe, but I don't have a clear picture of how we use the feed types. If it's just one check somewhere in `rss_parser`, I wouldn't bother. OTOH, if we systematically check the type to treat feeds differently, I'd go for enum or even a class hierarchy if it makes sense (again, need to understand the usage).

Not ready to go code diving right now to find all of this out. Any volunteers? :)

File: test/rss.cpp (hunk 1 above)
Comment (kmws) [reply test/rss.cpp]: I would probably leave that to you as a thing to find out... and write a proper description. ;)

Sure, I will change them.

**Comment by kmws** [comment]
>perhaps an enum would be more appropriate here?

Probably what @Minoru said, it needs to be checked out whether it's worth the hassle now. I can do that, but not today.

**Comment by tsipinakis** [comment]
Did I ever say how much I love grep? :p 

```
$ grep -r -e '"filter:"' -e '"exec:"' src/
src/utils.cpp:	return url.substr(0, 6) == "query:" || url.substr(0, 7) == "filter:" ||
src/utils.cpp:		url.substr(0, 5) == "exec:";
src/rss_parser.cpp:	} else if (uri.substr(0, 5) == "exec:") {
src/rss_parser.cpp:	} else if (uri.substr(0, 7) == "filter:") {
```

There are 2 functions where we use the equivalent checks, `rss_parser::retrieve_uri` and `utils::is_special_url`. Looks like @Minoru was right, most likely not worth the hassle for 4 references.

**Comment by Minoru** [comment]
Hmm, I expected that to be much harder than that (dunno why I did tho). Thank you, @tsipinakis! BTW, try `git grep` sometime: it's faster, requires less arguments, and doesn't search in files not tracked by Git (so no results from *.o and .git/).

**Comment by kmws** [comment]
Hmm, I would still make new util functions for "exec:" and "filter:" to be consistent, even if they have one case each.

**Comment by Minoru** [comment]
@kmws go for it, then! I'll merge tomorrow.

**Comment by Minoru** [comment]
Awesome! Thanks a lot, @kmws!

**Pull request merged**